src/pages/ResearcherProfile.js

```diff
@@ -1,492 +1,351 @@
+      try {
+        setIsLoading(true);
+        await getResearcherProfile();
+
+        setInstitutionList(await Institution.list());
+        generateCountryNames();
+        generateStateNames();
+
+        props.history.push('profile');
+        setNotificationData(await NotificationService.getBannerObjectById('eRACommonsOutage'));
+
+        setCurrentUser(Storage.getCurrentUser());
+        setIsLoading(false);
+      } catch (error) {
+        Notification.showError({text: 'Error: Unable to retrieve user data from server'});
+      } finally {
+        setIsLoading(false);
+      }
     };
-  }
 
-  getResearcherProfile = async () => {
-    const user = await User.getMe();
-    const researcherProps = getPropertyValuesFromUser(user);
-    const institutionList = await Institution.list();
-
-    this.setState(prev => {
-      if (isEmpty(user.roles)) {
-        prev.roles = [{ 'roleId': 5, 'name': USER_ROLES.researcher }];
-      } else {
-        prev.roles = user.roles;
-      }
-      prev.additionalEmail = isNil(user.additionalEmail) ? '' : user.additionalEmail;
-      prev.institutionId = user.institutionId;
-      prev.institutionList = institutionList;
-      prev.profile.academicEmail = researcherProps.academicEmail || user.email;
-      prev.profile.address1 = researcherProps.address1;
-      prev.profile.address2 = researcherProps.address2;
-      prev.profile.checkNotifications = (researcherProps.checkNotifications === 'true');
-      prev.profile.city = researcherProps.city;
-      prev.profile.completed = researcherProps.completed;
-      prev.profile.country = researcherProps.country;
-      prev.profile.department = researcherProps.department;
-      prev.profile.division = researcherProps.division;
-      prev.profile.eRACommonsID = researcherProps.eraCommonsId;
-      prev.profile.havePI = researcherProps.havePI;
-      prev.profile.isThePI = researcherProps.isThePI;
-      prev.profile.linkedIn = researcherProps.linkedIn;
-      prev.profile.orcid = researcherProps.orcid;
-      prev.profile.piEmail = researcherProps.piEmail;
-      prev.profile.piName = researcherProps.piName;
-      prev.profile.profileName = user.displayName;
-      prev.profile.pubmedID = researcherProps.pubmedID;
-      prev.profile.researcherGate = researcherProps.researcherGate;
-      prev.profile.scientificURL = researcherProps.scientificURL;
-      prev.profile.state = researcherProps.state;
-      prev.profile.zipcode = researcherProps.zipcode;
-      return prev;
-    }, () => {
-      if (this.state.profile.completed !== undefined && this.state.profile.completed !== '') {
-        this.setState(prev => {
-          prev.profile.completed = JSON.parse(this.state.profile.completed);
-          return prev;
-        });
-      }
-    });
-  };
+    init();
+  }, [props.history]);
 
-  handleChange = (event) => {
-    let field = event.target.name;
-    let value = event.target.value;
+  useEffect(() => {
+    const isValid = (value) => {
```

Comment:
The linter recommended I move those functions into useEffect. They don't get used outside of the incomplete checks so I didn't think it would hurt.

Reply:
They aren't used outside of the useEffect, that's true, and it's correct that it has no effect on functionality either inside or outside. The linter warning is very strange ... these methods have no dependency on any state values - they're purely functional and could be in an external library if we had more use for them. For example,
```
    const isValidNumber = (value) => {
      let isValid = false;
      if (value !== 0 && value !== null && value !== undefined) {
        isValid = true;
      }
      return isValid;
    };
``` 
Actually can be a call to the lodash function just as easily, so I'm really confused about why the linter is complaining. The point of my comment is that as more things are moved inside of useEffect, it becomes harder to reason about. The things that really need to be there are the state-affecting functions that useEffect has a dependency on, i.e. things that touch `profile` and `incompletes`

Reply:
I'm guessing `isValidState` has to be included because it references the state variable `profile` whose value can change, which is why the linter is stating that it needs to be included. However I don't see linter warnings for the other isValid methods, so those can safely be declared outside the useEffect.

Reply:
Figured out and fixed